mypy job failure in CI workflow due to types-pkg_resources

When the CI workflow runs on GitHub Actions, the `mypy` job fails with the following error:
```
arkouda/client.py:3: error: Library stubs not installed for "pkg_resources" (or incompatible with Python 3.7)
arkouda/client.py:3: note: Hint: "python3 -m pip install types-pkg_resources"
arkouda/client.py:3: note: (or run "mypy --install-types" to install all missing stub packages)
arkouda/client.py:3: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 19 source files)
Error: Process completed with exit code 1.
```

This isn't an issue with our code, rather it appears it is a missing sub-package which isn't being installed in the container for whatever reason.

Adding the missing package to the workflow mypy job, run section solves the issue on my fork when running the CI workflow Action.

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -23,6 +23,7 @@ jobs:
     - name: Install dependencies
       run: |
         apt-get update && apt-get install -y python3-pip
+        python3 -m pip install types-pkg_resources
         python3 -m pip install -e .[dev]
     - name: Arkouda mypy
       run: |

